### PR TITLE
add cumulative sum diagnostic

### PR DESCRIPTION
**Pull request by 12shughes**
Added new diagnostic that calculates a cumulative sum of a field across every timestep

### CODE REVIEW

File: gusto/diagnostics/diagnostics.py [hunk 1]
```diff
@@ -1040,3 +1041,42 @@ def setup(self, domain, state_fields):
...
+class CumulativeSum(DiagnosticField):
+    """Base diagnostic for cumulatively summing a field in time."""
+    def __init__(self, name):
+        """
+        Args:
+            name (str): name of the field to take the cumulative sum of.
+        """
+        self.field_name = name
+        self.integral_name = name+"_cumulative"
+        DiagnosticField.__init__(self, method='assign', required_fields=(self.field_name,))
```
Comment (jshipton) [inline gusto/diagnostics/diagnostics.py:1055]:
```suggestion
        super().__init__(self, method='assign', required_fields=(self.field_name,))
```

**Review by jshipton — CHANGES_REQUESTED** [review]
one tiny change (and a test that I think @tommbendall is talking to you about right now)

File: gusto/diagnostics/diagnostics.py [hunk 2]
```diff
@@ -1040,3 +1041,42 @@ def setup(self, domain, state_fields):
...
+            state_fields (:class:`StateFields`): the model's field container.
+        """
+
+        # Gather the field to be summed
+        self.integrand = state_fields(self.field_name)
+        self.space = self.integrand.function_space()
+
+        # Create a new field to store the cumulative sum
+        self.field = state_fields(self.integral_name, space=self.space, dump=True, pick_up=True)
+        # Initialise the new field to zero, if picking up from a checkpoint file the original cumulative field will load and not be overwritten.
```
[the remark this answers is not in the document]
Comment (12shughes) [reply gusto/diagnostics/diagnostics.py]: This is now done

**Review by tommbendall — APPROVED** [review]
Brilliant, thanks @12shughes

**Pull request merged**